### DISCUSSION TITLE
Web3Auth Flutter SDK

### DISCUSSION THREAD

**Discussion by adityathakurxd** [Web3Auth/Web3Auth#575] [Help]
Had some issues in integrating the Web3Auth Flutter SDK at the Hacker House Dilli. Listing all the points and issues below:

**1) The installation points to a now unavailable page:**
What it is: https://github.com/torusresearch/web3auth-flutter-sdk.git
What it should be: https://github.com/web3auth/web3auth-flutter-sdk

Can also maybe add directly adding `web3auth_flutter: ^1.0.0` to `pubspec.yaml`.

**2) The `initPlatformState` function should be updated on the [initialize page](https://web3auth.io/docs/sdk/flutter/initialize):**
```dart
Future<void> initPlatformState() async {
    // Initialization
    await Web3AuthFlutter.init(
      Web3AuthOptions(
        clientId:
            "YOUR WEB3AUTH CLIENT ID FROM DASHBOARD",
        network: Network.testnet,
      ),
    );
  }
```

3) Similarly the Web3AuthResponse doc 
```dart
final Web3AuthResponse response = await Web3AuthFlutter.login(
                LoginParams(loginProvider: Provider.google));
```

4) Lastly, getting an error when running the project

This is probably due to some Android configuration issues.

**Error:** 
<img width="680" alt="Screenshot 2022-09-08 at 4 33 40 PM" src="https://user-images.githubusercontent.com/53579386/189106496-73d3d71d-e45e-4672-9ee9-b45116819319.png">
cc @shahbaz17

**Comment by shahbaz17** [comment]
Hey @adityathakurxd 

Let's meet during the hackathon. I got this to work.

Meanwhile, can you share your `flutter doctor` output here?

**Comment by adityathakurxd** [comment]
Thank you @shahbaz17! I redid the project and used Android Studio to better debug. It is working now on my system.
There is this one issue happening, I can come by your table at the Hackathon whenever you're available. 

![WhatsApp Image 2022-09-09 at 11 44 57 AM (1)](https://user-images.githubusercontent.com/53579386/189283884-12edd04a-4b10-433e-af83-ae74ac76590e.jpeg)

It does launch an in-app browser to login although the 'Redirect to app' doesn't work once auth is done.

**Reply by shahbaz17** [reply]
Can you retry installing the app using `flutter run` or through Android Studio?

**Reply by adityathakurxd** [reply]
I have tried restarting the app. It doesn't return to the app after login and the result is empty.

**Comment by shahbaz17** [comment]
The [latest doc on Flutter](https://web3auth.io/docs/sdk/flutter/) resolves all of the issues stated in this discussion.
